src/rgw/rgw_asio_frontend.cc

```diff
@@ -279,15 +279,17 @@ void handle_connection(boost::asio::io_context& context,
       }
       int http_ret = 0;
       string user = "-";
+      string bucket = "-";
       const auto started = ceph::coarse_real_clock::now();
       ceph::coarse_real_clock::duration latency{};
       process_request(env, &req, uri_prefix, &client, y,
-                      scheduler, &user, &latency, &http_ret);
+                      scheduler, &user, &bucket, &latency, &http_ret);
 
       if (cct->_conf->subsys.should_gather(ceph_subsys_rgw_access, 1)) {
         // access log line elements begin per Apache Combined Log Format with additions following
         lsubdout(cct, rgw_access, 1) << "beast: " << std::hex << &req << std::dec << ": "
-            << remote_endpoint.address() << " - " << user << " [" << log_apache_time{started} << "] \""
+            << remote_endpoint.address() << " - " << user << " - " << bucket
+            << " [" << log_apache_time{started} << "] \""
```

Comment:
these log lines are intended to be in 'Apache Combined Log Format', so we shouldn't be adding new fields in the middle

cc @mkogan1 who added this in https://github.com/ceph/ceph/pull/33083

Reply:
>have you considered using the [rgw ops log](https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_enable_ops_log) for this use case?

Let us check this one out.

> these log lines are intended to be in 'Apache Combined Log Format', so we shouldn't be adding new fields in the middle

Ahh, that makes sense. AFAIU, though, we can add it to the end (like `latency` field in the end) - if we decide to add bucket as an explicit field, right?

Reply:
> > have you considered using the [rgw ops log](https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_enable_ops_log) for this use case?
> 
> Let us check this one out.

when enabled, it will write json output to rgw_ops_log_file_path. the json format contains all the fields listed in https://github.com/ceph/ceph/blob/c2bd611/src/rgw/rgw_log.cc#L250-L350

Reply:
> Ahh, that makes sense. AFAIU, though, we can add it to the end (like `latency` field in the end) - if we decide to add bucket as an explicit field, right?

that sounds reasonable to me

we might also consider adding the bucket name to the front of the path for those vhost-style requests, but that would be more work and potentially cause some confusion

Reply:
> have you considered using the [rgw ops log](https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_enable_ops_log) for this use case?
> when enabled, it will write json output to rgw_ops_log_file_path. the json format contains all the fields listed in

This looks really promising. It has some missing fields like latency I think but I'll take a closer look later tomorrow.

>we can add it to the end (like latency field in the end)

Added a new evidence to correct the problem; now bucket field is added right before the latency field in the end. I believe/hope this abides by the Apache Combined Log format.

> we might also consider adding the bucket name to the front of the path for those vhost-style requests, but that would be more work and potentially cause some confusion

The biggest problem is, to the best of our knowledge, there's no definitive way to tell whether an HTTP S3 request uses virtual-host style or not. We thought of searching `bucket` name in `message["host"]` to decide but this is not a generic solution since a domain name can be anything and can contain the word a client is trying to access (the bucket).

We can amend the path as well (e.g.,, "PUT <here> <message.target>".  Let's see what @mkogan1 think as well.

Reply:
@cbodley , @mkogan1 - we have started using _rgw-ops-log_ but as described in the [Description](https://github.com/ceph/ceph/pull/50350#issue-1607176983) (section #Testing with RGW Ops Log) and in the second commit (https://github.com/ceph/ceph/pull/50350/commits/76cf80c18d87941d50c8c41d807993b6a6ad5026), in certain cases when virtual-host access style is used, there's still an ambiguity when trying to figure out `bucket` and `object` from log lines. The 2nd commit adds `object` as well both to _beast_ and _ops-log_ to address it.